### ISSUE TITLE
mx.metric.CrossEntropy/NegativeLogLikelihood is using numpy logic and partially duplicates mx.gluon.loss.SoftmaxCELoss

### ISSUE THREAD

**Issue by szha** [apache/incubator-mxnet#9593]
metric.CrossEntropy: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/metric.py#L909-L918
metric.NegativeLogLikelihood: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/metric.py#L977-L986
gluon.loss.SoftmaxCELoss: https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/loss.py#L310-L318
